tests/NextButton.test.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import { NextButton } from '../src/NextButton';
+
+const mockOnClick = jest.fn();
+
+const defaultProps = {
+    onClick: mockOnClick,
+};
+
+const renderWithProps = (props = {}) => render(<NextButton { ...defaultProps } { ...props }>foo</NextButton>);
+
+describe('NextButton Component', () => {
+    it('should render correctly', () => {
+        const { asFragment } = renderWithProps();
+
+        expect(asFragment()).toMatchSnapshot();
+    });
+
+    it('should render correctly when passing href', () => {
+        const { asFragment } = renderWithProps({ href: 'bar' });
+
+        expect(asFragment()).toMatchSnapshot();
+    });
+
+    it('should render correctly when passing onClick', () => {
```

Comment:
Reading this test description seems that we are only testing if it's being rendered correctly. However, we're checking that `onClick` is being fired correctly.

What do you think of this?
> should call onClick correctly when passed